packages/adaptive-ui/src/core/color/palette-base.ts

```diff
@@ -1,17 +1,16 @@
 import { Color } from "./color.js";
 import { Palette, PaletteDirection, PaletteDirectionValue, resolvePaletteDirection } from "./palette.js";
-import { Swatch } from "./swatch.js";
 import { binarySearch } from "./utilities/binary-search.js";
 import { directionByIsDark } from "./utilities/direction-by-is-dark.js";
 import { contrast, RelativeLuminance } from "./utilities/relative-luminance.js";
 
 /**
  * A base {@link Palette} with a common implementation of the interface. Use PaletteRGB for an implementation
- * of a palette generation algorithm that is ready to be used directly, or extend this class to generate custom Swatches.
+ * of a palette generation algorithm that is ready to be used directly, or extend this class to generate custom swatches.
  *
  * @public
  */
-export class BasePalette<T extends Swatch> implements Palette<T> {
+export class BasePalette<T extends Color = Color> implements Palette<T> {
```

Comment:
Curious on your thoughts here. Right now Palette only supports solid Colors, but I believe a Gradient Palette (that is, a palette of Gradient swatches) still fits in the model. Each swatch would be a Gradient, say a hue rotation, with the same luminance levels as if they were solid colors.

It wouldn't break anything to change this to `Paint`, though I'll probably want to do that as a follow-up as I mention in the PR description.

Reply:
I think it can make sense, though I generally think of palettes as sets of solid colors. I don't know that that thinking is for any reason other than convention / past experience though.

Reply:
Cool, that aligns with my thoughts. Probably one of the primary distinctions that becomes less clear when introducing anything _other_ than a solid color is our reliance on luminance. I have addressed that now in that both gradient algorithms maintain a consistent luminance as much as possible. Some potential treatments won't do that, so perhaps we need to relax the system or at least account for places where we don't have a confident single value.